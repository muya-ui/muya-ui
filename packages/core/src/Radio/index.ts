export { default } from './Radio';
export { default as Radio } from './Radio';
export { default as RadioGroup } from './RadioGroup';
export { default as useRadio } from './useRadio';
export { default as useRadioGroup } from './useRadioGroup';

export * from './types';
