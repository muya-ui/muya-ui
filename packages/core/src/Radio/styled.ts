import styled, { css } from 'styled-components';

import { IComponentSizeSpec, IFontSizeSpec } from '@muya-ui/theme-light';

import { ellipsisWithWidth } from '../styled/mixins/ellipsis';
import { IThemedBaseProps } from '../types';

export interface IStyledRadioInnerProps extends IThemedBaseProps {
  $size: IComponentSizeSpec;
  $checked: boolean;
}

export interface IStyledTextProps extends IThemedBaseProps {
  $size: IComponentSizeSpec;
}

export interface IStyledRadioLabelProps extends IThemedBaseProps {
  $disabled: boolean;
  $checked: boolean;
  $width?: string | number;
  $ellipsis: boolean;
}

export const StyledRadioInput = styled.input`
  box-sizing: border-box;
  touch-action: manipulation;
  position: absolute;
  z-index: 1;
  cursor: pointer;
  opacity: 0;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
`;

export const StyledRadioIcon = styled.span`
  position: relative;
  display: inline-block;
  line-height: 1;
  white-space: nowrap;
  vertical-align: middle;
  outline: none;
  cursor: pointer;
`;

const radioInnerCSS = (props: IStyledRadioInnerProps) => {
  const {
    theme: {
      components: { Radio: token },
      transition: {
        spec: { duration, easing },
      },
    },
    $size,
    $checked,
  } = props;
  const iconSize = token.iconSize[$size!];
  const checkedCenterSize = token.checkedCenterSize[$size!];

  return css`
    transition: all ${duration.normal}ms ${easing.easeInOut};
    width: ${iconSize}px;
    height: ${iconSize}px;
    border-width: 1px;
    border-color: ${$checked ? token.borderColor.checked : token.borderColor.unChecked};
    background-color: ${$checked ? token.iconColor.checked : token.iconColor.unChecked};
    &::after {
      content: '';
      width: ${checkedCenterSize}px;
      height: ${checkedCenterSize}px;
      background: ${token.checkedCenterBg};
      transform: ${$checked ? 'scale(1)' : 'scale(0)'};
      position: absolute;
      border-radius: 50%;
      top: ${(iconSize - checkedCenterSize) / 2}px;
      left: ${(iconSize - checkedCenterSize) / 2}px;
      transition: all ${duration.normal}ms ${easing.easeInOut};
    }
  `;
};

export const StyledRadioInner = styled.span`
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  border-style: solid;
  ${radioInnerCSS}
`;

const textSizeCSS = (props: IStyledTextProps) => {
  const {
    theme: {
      colors,
      typography: {
        spec: { fontSize, lineHeight },
      },
      spacing: {
        pattern: { textIcon },
      },
      components: { Radio: token },
    },
    $size,
  } = props;
  const fontLevel = token.fontLevel[$size!] as IFontSizeSpec;
  return css`
    color: ${colors.pattern.text.text};
    font-size: ${fontSize[fontLevel]}px;
    line-height: ${lineHeight[fontLevel]}px;
    padding-left: ${textIcon[fontLevel]}px;
    padding-right: ${textIcon[fontLevel]}px;
  `;
};

export const StyledRadioText = styled.span`
  vertical-align: middle;
  ${textSizeCSS}
`;

// 状态的样式
const radioStatusCSS = (props: IStyledRadioLabelProps) => {
  const {
    theme: {
      opacity,
      components: { Radio: token },
    },
    $disabled,
    $checked,
    $width,
    $ellipsis,
  } = props;

  const interactive = !$disabled;
  let interactiveCSS;
  if (!interactive) {
    interactiveCSS = css`
      opacity: ${opacity.pattern.disabled};
      cursor: not-allowed;
      ${StyledRadioInput}, ${StyledRadioIcon} {
        cursor: not-allowed;
      }
    `;
  } else {
    interactiveCSS = css`
      &:focus,
      &:hover {
        ${StyledRadioInner} {
          border-color: ${$checked ? token.borderColor.checkedHover : token.borderColor.hover};
          background-color: ${$checked ? token.iconColor.checkedHover : token.iconColor.hover};
        }
      }
      &:active {
        ${StyledRadioInner} {
          border-color: ${$checked ? token.borderColor.checkedClick : token.borderColor.click};
          background-color: ${$checked ? token.borderColor.checkedClick : token.iconColor.click};
        }
      }
    `;
  }

  return css`
    ${ellipsisWithWidth($ellipsis, $width)}
    ${interactiveCSS}
  `;
};

export const StyledRadioLabel = styled.label`
  box-sizing: border-box;
  user-select: none;
  display: inline-block;
  white-space: nowrap;
  cursor: pointer;
  ${radioStatusCSS}
`;

export const StyledRadioGroup = styled.div<IThemedBaseProps>`
  display: inline-block;
  ${StyledRadioLabel} {
    margin-right: ${props => props.theme.components.Radio.marginRightInGroup}px;
  }
`;
