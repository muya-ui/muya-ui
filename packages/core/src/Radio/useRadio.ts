import React, { ChangeEvent, useState, useCallback } from 'react';

import { IRadioProps } from './types';

export default function useRadio<T extends HTMLElement>(props: IRadioProps, isControlled: boolean) {
  const { checked, defaultChecked = false, disabled, onChange, onClick } = props;

  const [checkedState, setCheckedState] = useState(isControlled ? checked : defaultChecked);

  const handleInputChange = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      if (disabled) {
        return;
      }
      const value = e.target.checked;
      if (!isControlled) {
        setCheckedState(value);
      }
      if (onChange) {
        onChange(e);
      }
    },
    [disabled, isControlled, onChange],
  );

  const handleInputClick = useCallback((e: React.MouseEvent) => {
    e.preventDefault();
    e.stopPropagation();
    if (e.nativeEvent) {
      e.nativeEvent.stopImmediatePropagation();
    }
  }, []);

  const handleClick = useCallback(
    (e: React.MouseEvent<T, MouseEvent>) => {
      if (disabled) {
        return;
      }
      // if (!isControlled) {
      //   setCheckedState(!checkedState);
      // }
      if (onClick) {
        onClick(e);
      }
    },
    [disabled, onClick],
  );

  if (isControlled && checkedState !== props.checked) {
    setCheckedState(props.checked!);
  }

  return {
    checkedState,
    handleInputChange,
    handleInputClick,
    handleClick,
    setCheckedState,
  };
}
