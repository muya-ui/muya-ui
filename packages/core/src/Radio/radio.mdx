---
name: Radio 单选
menu: 组件
route: /components/radio
autoDemos: true
parent: 数据录入
---

import { DocRadioStyle } from './demos/styled';

<DocRadioStyle />

# Radio 单选

用于多个选项中选择一个选项。

## 使用场景

- 适用于默认可见，选项不宜过多。
- 若所有选项之间有线性关系，推荐使用滑动条、评分组件；若选项是两种互斥状态的切换，如开关功能，推荐使用开关。

## 组件

- `Radio`：单选。
- `RadioGroup`：单选组。

{Demos}

### API

#### IRadioProps

<Props propsName="IRadioProps" />

#### IRadioGroupProps

<Props propsName="IRadioGroupProps" />

#### IRadioGroupOption

<Props propsName="IRadioGroupOption" omit={['onChange']} />

#### IRadioGroupValue

```ts
type IRadioGroupValue = string | number | null;
```

### Styles

#### IRadioStyleKeys

<Styles typeName="IRadioStyleKeys" componentName="radio" />
