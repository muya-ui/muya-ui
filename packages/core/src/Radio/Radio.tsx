import React, { Ref } from 'react';

import memoForwardRef from '../utils/memoForwardRef';
import useStyles from '../utils/useStyles';
import useTheme from '../utils/useTheme';
import {
  StyledRadioIcon,
  StyledRadioInner,
  StyledRadioInput,
  StyledRadioLabel,
  StyledRadioText,
} from './styled';
import { IRadioProps } from './types';
import useRadio from './useRadio';

const defaultStyles = {
  wrapper: '',
  inputRadio: '',
  innerRadio: '',
  text: '',
};

const Radio = memoForwardRef((props: IRadioProps, ref: Ref<HTMLLabelElement>) => {
  const {
    autoFocus = false,
    disabled = false,
    defaultChecked = false,
    ellipsis = false,
    width,
    checked,
    children,
    size = 'm',
    onChange,
    onMouseEnter,
    onMouseLeave,
    styles,
    inputRef,
    className,
    style,
    ...restProps
  } = props;
  const theme = useTheme();
  const innerStyles = useStyles('radio', defaultStyles, styles);
  const { checkedState, handleInputChange, handleClick, handleInputClick } = useRadio(
    props,
    'checked' in props,
  );

  return (
    <StyledRadioLabel
      ref={ref}
      theme={theme}
      $disabled={disabled}
      $width={width}
      $ellipsis={ellipsis}
      $checked={checkedState!}
      className={[innerStyles.wrapper.className, className].join(' ').trim()}
      style={{ ...innerStyles.wrapper.style, ...style }}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
      onClick={handleClick}
    >
      <StyledRadioIcon>
        <StyledRadioInput
          {...restProps}
          {...innerStyles.inputRadio}
          ref={inputRef}
          type="radio"
          autoFocus={autoFocus}
          checked={checkedState}
          onClick={handleInputClick}
          onChange={handleInputChange}
          disabled={disabled}
        />
        <StyledRadioInner
          {...innerStyles.innerRadio}
          theme={theme}
          $checked={checkedState!}
          $size={size}
        />
      </StyledRadioIcon>
      <StyledRadioText {...innerStyles.text} theme={theme} $size={size}>
        {children}
      </StyledRadioText>
    </StyledRadioLabel>
  );
});

export default Radio;
