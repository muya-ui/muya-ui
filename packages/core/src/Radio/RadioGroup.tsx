import React, { Ref, useMemo } from 'react';

import memoForwardRef from '../utils/memoForwardRef';
import useTheme from '../utils/useTheme';
import Radio from './Radio';
import { StyledRadioGroup } from './styled';
import { IRadioGroupOption, IRadioGroupProps } from './types';
import useRadioGroup from './useRadioGroup';

const RadioGroup = memoForwardRef((props: IRadioGroupProps, ref: Ref<HTMLDivElement>) => {
  const {
    opts,
    valueState,
    handleChange,
    handleClick,
    children,
    renderRadio,
    radioEllipsis,
    radioWidth,
    restProps,
  } = useRadioGroup(props);
  const theme = useTheme();
  const radioList = useMemo(() => {
    if (opts && opts.length) {
      return opts.map((option, optIndex) => {
        const { label, value, disabled, ellipsis, width } = option;
        const radioNode = (
          <Radio
            key={String(value)}
            value={value}
            size={props.size}
            styles={props.styles}
            ellipsis={ellipsis === undefined ? radioEllipsis : ellipsis}
            width={width || radioWidth}
            disabled={props.disabled || disabled}
            checked={valueState === value}
            onChange={handleChange.bind(null, option)}
            onClick={handleClick.bind(null, option)}
          >
            {label}
          </Radio>
        );
        return renderRadio(radioNode, option, optIndex);
      });
    }

    return React.Children.map(children, (child, childIndex) => {
      if (!React.isValidElement(child)) {
        return null;
      }
      const {
        value,
        disabled,
        onChange,
        children: label,
        ellipsis,
        width,
        ...otherChildProps
      } = child.props;
      const option: IRadioGroupOption = { value, disabled, onChange, label };
      const radioNode = React.cloneElement(child, {
        ...otherChildProps,
        children: label,
        value,
        ellipsis: ellipsis === undefined ? radioEllipsis : ellipsis,
        width: width || radioWidth,
        checked: valueState === value,
        disabled: props.disabled || disabled,
        styles: props.styles,
        size: props.size,
        onChange: handleChange.bind(null, option),
        onClick: handleClick.bind(null, option),
      });
      return renderRadio(radioNode, option, childIndex);
    });
  }, [
    opts,
    children,
    props.size,
    props.styles,
    props.disabled,
    radioEllipsis,
    radioWidth,
    valueState,
    handleChange,
    handleClick,
    renderRadio,
  ]);

  return (
    <StyledRadioGroup ref={ref} theme={theme} {...restProps}>
      {radioList}
    </StyledRadioGroup>
  );
});

export default RadioGroup;
