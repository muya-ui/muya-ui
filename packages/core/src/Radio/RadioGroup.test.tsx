import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import { IRadioGroupValue } from '@muya-ui/core';

import Radio from './Radio';
import RadioGroup from './RadioGroup';

test('shound onChange ok when options is string[]', () => {
  const options = ['a', 'b', 'c'];
  const onChange = sinon.spy();
  let groupValue: IRadioGroupValue = null;
  const wrap = mount(
    <RadioGroup
      options={options}
      value={groupValue}
      onChange={(value: IRadioGroupValue) => {
        groupValue = value;
        onChange();
      }}
    />,
  );
  wrap
    .find('input')
    .first()
    .simulate('change', { target: { checked: true } });
  expect(groupValue).toEqual('a');
  expect(onChange.calledOnce).toBe(true);
});

test('shound defaultValue take effect when options is object[]', () => {
  const onChange1 = sinon.stub();
  const onChange2 = sinon.stub();
  const options = [
    { label: 'a', value: 1, onChange: onChange1 },
    { label: 'b', value: 2, onChange: onChange2 },
    { label: 'c', value: 3 },
    { label: 'd', value: 4, disabled: true },
  ];

  const wrap = mount(<RadioGroup options={options} defaultValue={2} />);
  expect(
    wrap
      .find('div')
      .childAt(1)
      .find('input')
      .prop('checked'),
  ).toEqual(true);
  wrap
    .find('input')
    .first()
    .simulate('change', { target: { checked: true } });
  expect(() => {
    sinon.assert.calledOnce(onChange1);
    sinon.assert.notCalled(onChange2);
  }).not.toThrow();
  wrap
    .find('div')
    .childAt(1)
    .find('input')
    .simulate('change', { target: { checked: false } });
  expect(() => {
    sinon.assert.calledOnce(onChange2);
  }).not.toThrow();
});

test('shound not change when all disabled', () => {
  const options = ['a', 'b', 'c'];
  const onChange = sinon.stub();
  const wrapper = mount(<RadioGroup options={options} value="b" disabled onChange={onChange} />);

  wrapper
    .find('input')
    .first()
    .simulate('change', { target: { checked: true } });
  wrapper
    .find('label')
    .first()
    .simulate('click');
  expect(() => {
    sinon.assert.notCalled(onChange);
  }).not.toThrow();
});

test('Snapshots should be matched in the base case', () => {
  const options = [
    { label: 'a', value: 1 },
    { label: 'b', value: 2 },
    { label: 'c', value: 3 },
    { label: 'd', value: 4, disabled: true },
  ];
  const tree = renderer.create(<RadioGroup options={options} defaultValue={2} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Snapshots 自定义 radio 渲染', () => {
  const tree = renderer
    .create(
      <RadioGroup defaultValue={2} renderRadio={node => <div>{node}</div>}>
        11
        <Radio value={1}>1</Radio>
        <Radio value={2}>2</Radio>
        <Radio value={3}>3</Radio>
      </RadioGroup>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Snapshots 自定义 width 和 ellipsis', () => {
  const tree = renderer
    .create(
      <RadioGroup defaultValue={2} radioEllipsis radioWidth={100}>
        11
        <Radio value={1}>1</Radio>
        <Radio value={2}>2</Radio>
        <Radio value={3}>3</Radio>
      </RadioGroup>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
