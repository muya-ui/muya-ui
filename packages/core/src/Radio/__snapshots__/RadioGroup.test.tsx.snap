// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should be matched in the base case 1`] = `
.c6 {
  box-sizing: border-box;
  touch-action: manipulation;
  position: absolute;
  z-index: 1;
  cursor: pointer;
  opacity: 0;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
}

.c4 {
  position: relative;
  display: inline-block;
  line-height: 1;
  white-space: nowrap;
  vertical-align: middle;
  outline: none;
  cursor: pointer;
}

.c8 {
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  border-style: solid;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  width: 16px;
  height: 16px;
  border-width: 1px;
  border-color: rgba(204,204,204,1);
  background-color: rgba(255,255,255,1);
}

.c8::after {
  content: '';
  width: 4px;
  height: 4px;
  background: rgba(255,255,255,1);
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  position: absolute;
  border-radius: 50%;
  top: 6px;
  left: 6px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
}

.c11 {
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  border-style: solid;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  width: 16px;
  height: 16px;
  border-width: 1px;
  border-color: rgba(26,122,248,1);
  background-color: rgba(26,122,248,1);
}

.c11::after {
  content: '';
  width: 4px;
  height: 4px;
  background: rgba(255,255,255,1);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  position: absolute;
  border-radius: 50%;
  top: 6px;
  left: 6px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
}

.c9 {
  vertical-align: middle;
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  padding-left: 4px;
  padding-right: 4px;
}

.c2 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  white-space: nowrap;
  cursor: pointer;
}

.c2:focus .c7,
.c2:hover .c7 {
  border-color: rgba(26,122,248,1);
  background-color: rgba(255,255,255,1);
}

.c2:active .c7 {
  border-color: rgba(12,108,235,1);
  background-color: rgba(255,255,255,1);
}

.c10 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  white-space: nowrap;
  cursor: pointer;
}

.c10:focus .c7,
.c10:hover .c7 {
  border-color: rgba(51,139,255,1);
  background-color: rgba(51,139,255,1);
}

.c10:active .c7 {
  border-color: rgba(12,108,235,1);
  background-color: rgba(12,108,235,1);
}

.c12 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  white-space: nowrap;
  cursor: pointer;
  opacity: 0.3;
  cursor: not-allowed;
}

.c12 .c5,
.c12 .c3 {
  cursor: not-allowed;
}

.c0 {
  display: inline-block;
}

.c0 .c1 {
  margin-right: 8px;
}

<div
  className="c0"
>
  <label
    className="c1 c2 muya-radio-wrapper"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="c3 c4"
    >
      <input
        autoFocus={false}
        checked={false}
        className="c5 c6 muya-radio-input-radio"
        disabled={false}
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value={1}
      />
      <span
        className="c7 c8 muya-radio-inner-radio"
      />
    </span>
    <span
      className="c9 muya-radio-text"
    >
      a
    </span>
  </label>
  <label
    className="c1 c10 muya-radio-wrapper"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="c3 c4"
    >
      <input
        autoFocus={false}
        checked={true}
        className="c5 c6 muya-radio-input-radio"
        disabled={false}
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value={2}
      />
      <span
        className="c7 c11 muya-radio-inner-radio"
      />
    </span>
    <span
      className="c9 muya-radio-text"
    >
      b
    </span>
  </label>
  <label
    className="c1 c2 muya-radio-wrapper"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="c3 c4"
    >
      <input
        autoFocus={false}
        checked={false}
        className="c5 c6 muya-radio-input-radio"
        disabled={false}
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value={3}
      />
      <span
        className="c7 c8 muya-radio-inner-radio"
      />
    </span>
    <span
      className="c9 muya-radio-text"
    >
      c
    </span>
  </label>
  <label
    className="c1 c12 muya-radio-wrapper"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="c3 c4"
    >
      <input
        autoFocus={false}
        checked={false}
        className="c5 c6 muya-radio-input-radio"
        disabled={true}
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value={4}
      />
      <span
        className="c7 c8 muya-radio-inner-radio"
      />
    </span>
    <span
      className="c9 muya-radio-text"
    >
      d
    </span>
  </label>
</div>
`;

exports[`Snapshots 自定义 radio 渲染 1`] = `
.c6 {
  box-sizing: border-box;
  touch-action: manipulation;
  position: absolute;
  z-index: 1;
  cursor: pointer;
  opacity: 0;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
}

.c4 {
  position: relative;
  display: inline-block;
  line-height: 1;
  white-space: nowrap;
  vertical-align: middle;
  outline: none;
  cursor: pointer;
}

.c8 {
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  border-style: solid;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  width: 16px;
  height: 16px;
  border-width: 1px;
  border-color: rgba(204,204,204,1);
  background-color: rgba(255,255,255,1);
}

.c8::after {
  content: '';
  width: 4px;
  height: 4px;
  background: rgba(255,255,255,1);
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  position: absolute;
  border-radius: 50%;
  top: 6px;
  left: 6px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
}

.c11 {
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  border-style: solid;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  width: 16px;
  height: 16px;
  border-width: 1px;
  border-color: rgba(26,122,248,1);
  background-color: rgba(26,122,248,1);
}

.c11::after {
  content: '';
  width: 4px;
  height: 4px;
  background: rgba(255,255,255,1);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  position: absolute;
  border-radius: 50%;
  top: 6px;
  left: 6px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
}

.c9 {
  vertical-align: middle;
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  padding-left: 4px;
  padding-right: 4px;
}

.c2 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  white-space: nowrap;
  cursor: pointer;
}

.c2:focus .c7,
.c2:hover .c7 {
  border-color: rgba(26,122,248,1);
  background-color: rgba(255,255,255,1);
}

.c2:active .c7 {
  border-color: rgba(12,108,235,1);
  background-color: rgba(255,255,255,1);
}

.c10 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  white-space: nowrap;
  cursor: pointer;
}

.c10:focus .c7,
.c10:hover .c7 {
  border-color: rgba(51,139,255,1);
  background-color: rgba(51,139,255,1);
}

.c10:active .c7 {
  border-color: rgba(12,108,235,1);
  background-color: rgba(12,108,235,1);
}

.c12 .c5,
.c12 .c3 {
  cursor: not-allowed;
}

.c0 {
  display: inline-block;
}

.c0 .c1 {
  margin-right: 8px;
}

<div
  className="c0"
>
  <div>
    <label
      className="c1 c2 muya-radio-wrapper"
      onClick={[Function]}
      style={Object {}}
    >
      <span
        className="c3 c4"
      >
        <input
          autoFocus={false}
          checked={false}
          className="c5 c6 muya-radio-input-radio"
          disabled={false}
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value={1}
        />
        <span
          className="c7 c8 muya-radio-inner-radio"
        />
      </span>
      <span
        className="c9 muya-radio-text"
      >
        1
      </span>
    </label>
  </div>
  <div>
    <label
      className="c1 c10 muya-radio-wrapper"
      onClick={[Function]}
      style={Object {}}
    >
      <span
        className="c3 c4"
      >
        <input
          autoFocus={false}
          checked={true}
          className="c5 c6 muya-radio-input-radio"
          disabled={false}
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value={2}
        />
        <span
          className="c7 c11 muya-radio-inner-radio"
        />
      </span>
      <span
        className="c9 muya-radio-text"
      >
        2
      </span>
    </label>
  </div>
  <div>
    <label
      className="c1 c2 muya-radio-wrapper"
      onClick={[Function]}
      style={Object {}}
    >
      <span
        className="c3 c4"
      >
        <input
          autoFocus={false}
          checked={false}
          className="c5 c6 muya-radio-input-radio"
          disabled={false}
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value={3}
        />
        <span
          className="c7 c8 muya-radio-inner-radio"
        />
      </span>
      <span
        className="c9 muya-radio-text"
      >
        3
      </span>
    </label>
  </div>
</div>
`;

exports[`Snapshots 自定义 width 和 ellipsis 1`] = `
.c6 {
  box-sizing: border-box;
  touch-action: manipulation;
  position: absolute;
  z-index: 1;
  cursor: pointer;
  opacity: 0;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
}

.c4 {
  position: relative;
  display: inline-block;
  line-height: 1;
  white-space: nowrap;
  vertical-align: middle;
  outline: none;
  cursor: pointer;
}

.c8 {
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  border-style: solid;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  width: 16px;
  height: 16px;
  border-width: 1px;
  border-color: rgba(204,204,204,1);
  background-color: rgba(255,255,255,1);
}

.c8::after {
  content: '';
  width: 4px;
  height: 4px;
  background: rgba(255,255,255,1);
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  position: absolute;
  border-radius: 50%;
  top: 6px;
  left: 6px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
}

.c11 {
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  border-style: solid;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  width: 16px;
  height: 16px;
  border-width: 1px;
  border-color: rgba(26,122,248,1);
  background-color: rgba(26,122,248,1);
}

.c11::after {
  content: '';
  width: 4px;
  height: 4px;
  background: rgba(255,255,255,1);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  position: absolute;
  border-radius: 50%;
  top: 6px;
  left: 6px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
}

.c9 {
  vertical-align: middle;
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  padding-left: 4px;
  padding-right: 4px;
}

.c12:focus .c7,
.c12:hover .c7 {
  border-color: rgba(26,122,248,1);
  background-color: rgba(255,255,255,1);
}

.c12:active .c7 {
  border-color: rgba(12,108,235,1);
  background-color: rgba(255,255,255,1);
}

.c13:focus .c7,
.c13:hover .c7 {
  border-color: rgba(51,139,255,1);
  background-color: rgba(51,139,255,1);
}

.c13:active .c7 {
  border-color: rgba(12,108,235,1);
  background-color: rgba(12,108,235,1);
}

.c14 .c5,
.c14 .c3 {
  cursor: not-allowed;
}

.c2 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  white-space: nowrap;
  cursor: pointer;
  width: 100px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c2:focus .c7,
.c2:hover .c7 {
  border-color: rgba(26,122,248,1);
  background-color: rgba(255,255,255,1);
}

.c2:active .c7 {
  border-color: rgba(12,108,235,1);
  background-color: rgba(255,255,255,1);
}

.c10 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  white-space: nowrap;
  cursor: pointer;
  width: 100px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c10:focus .c7,
.c10:hover .c7 {
  border-color: rgba(51,139,255,1);
  background-color: rgba(51,139,255,1);
}

.c10:active .c7 {
  border-color: rgba(12,108,235,1);
  background-color: rgba(12,108,235,1);
}

.c0 {
  display: inline-block;
}

.c0 .c1 {
  margin-right: 8px;
}

<div
  className="c0"
>
  <label
    className="c1 c2 muya-radio-wrapper"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="c3 c4"
    >
      <input
        autoFocus={false}
        checked={false}
        className="c5 c6 muya-radio-input-radio"
        disabled={false}
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value={1}
      />
      <span
        className="c7 c8 muya-radio-inner-radio"
      />
    </span>
    <span
      className="c9 muya-radio-text"
    >
      1
    </span>
  </label>
  <label
    className="c1 c10 muya-radio-wrapper"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="c3 c4"
    >
      <input
        autoFocus={false}
        checked={true}
        className="c5 c6 muya-radio-input-radio"
        disabled={false}
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value={2}
      />
      <span
        className="c7 c11 muya-radio-inner-radio"
      />
    </span>
    <span
      className="c9 muya-radio-text"
    >
      2
    </span>
  </label>
  <label
    className="c1 c2 muya-radio-wrapper"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="c3 c4"
    >
      <input
        autoFocus={false}
        checked={false}
        className="c5 c6 muya-radio-input-radio"
        disabled={false}
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value={3}
      />
      <span
        className="c7 c8 muya-radio-inner-radio"
      />
    </span>
    <span
      className="c9 muya-radio-text"
    >
      3
    </span>
  </label>
</div>
`;
